src/main/java/com/direwolf20/buildinggadgets/common/building/placement/SingleTypeProvider.java

```diff
@@ -0,0 +1,51 @@
+package com.direwolf20.buildinggadgets.common.building.placement;
+
+import com.direwolf20.buildinggadgets.common.building.IBlockProvider;
+import com.direwolf20.buildinggadgets.common.building.TranslationWrapper;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Immutable block provider that always give the same block state regardless of what position it requested.
```

Comment:
```suggestion
 * Immutable block provider that always returns the same block state regardless of which position is requested.
```
I'm unsure about the which/what thing - I'd need a Grammar book for that XD